pkg/apis/build/v1alpha1/build_validation_test.go

```diff
@@ -38,6 +38,369 @@ func TestValidateBuild(t *testing.T) {
 				}},
 			},
 		},
+	}, {
+		desc: "source and sources presence",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Name: "sources",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Source: &SourceSpec{
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+		reason: "source and sources cannot be declared in same build",
+	}, {
+		desc: "source without name",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+	}, {
+		desc: "source with targetPath",
+		build: &Build{
+			Spec: BuildSpec{
+				Sources: []SourceSpec{{
+					TargetPath: "/path/a/b",
+					Git: &GitSourceSpec{
+						Url:      "someurl",
+						Revision: "revision",
+					},
+				}},
+				Steps: []corev1.Container{{
+					Name:  "foo",
+					Image: "gcr.io/foo-bar/baz:latest",
+				}},
+			},
+		},
+	}, {
+		desc: "sources with duplicate targetPath",
```

Comment:
I'd really like to see us validate the core use case of checking out two github repos to their correct gopath location, aka something like:

```golang
build: &Build{
	Spec: BuildSpec{
		Sources: []SourceSpec{{
			TargetPath: "/go/src/github.com/knative/build",
			Git: &GitSourceSpec{
				Url:      "https://github.com/knative/build",
				Revision: "revision",
			},
		}, {
			TargetPath: "/go/src/github.com/knative/serving",
			Git: &GitSourceSpec{
				Url:      "https://github.com/knative/serving",
				Revision: "revision",
			},
		}},
		Steps: []corev1.Container{{
			Name:  "foo",
			Image: "gcr.io/foo-bar/baz:latest",
		}},
	},
},
reason: "sources with prefixed targetPath",
```

It isn't clear to me that this scenario will work since they both duplicate the /go/src/github.com/knative prefix

Reply:
That should work as the leaf dir is different.